x/feemarket/keeper/abci.go

```diff
@@ -58,25 +59,35 @@ func (k *Keeper) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) {
 		return
 	}
 
-	gasWanted := k.GetTransientGasWanted(ctx)
-	gasUsed := ctx.BlockGasMeter().GasConsumedToLimit()
+	gasWanted := sdkmath.NewIntFromUint64(k.GetTransientGasWanted(ctx))
+	gasUsed := sdkmath.NewIntFromUint64(ctx.BlockGasMeter().GasConsumedToLimit())
+
+	if !gasWanted.IsInt64() {
+		k.Logger(ctx).Error(fmt.Sprintf("integer overflow by integer type conversion. gasWanted %s is higher than MaxInt64", gasWanted))
+		return
+	}
+
+	if !gasUsed.IsInt64() {
+		k.Logger(ctx).Error(fmt.Sprintf("integer overflow by integer type conversion. gasUsed %s is higher than MaxInt64", gasUsed))
+		return
+	}
 
 	// to prevent BaseFee manipulation we limit the gasWanted so that
 	// gasWanted = max(gasWanted * MinGasMultiplier, gasUsed)
 	// this will be keep BaseFee protected from un-penalized manipulation
 	// more info here https://github.com/evmos/ethermint/pull/1105#discussion_r888798925
 	minGasMultiplier := k.GetParams(ctx).MinGasMultiplier
-	limitedGasWanted := sdk.NewDec(int64(gasWanted)).Mul(minGasMultiplier)
-	gasWanted = sdk.MaxDec(limitedGasWanted, sdk.NewDec(int64(gasUsed))).TruncateInt().Uint64()
-	k.SetBlockGasWanted(ctx, gasWanted)
+	limitedGasWanted := sdk.NewDec(gasWanted.Int64()).Mul(minGasMultiplier)
```

Comment:
Maybe you can transform to in64 once on top and reuse it so that the process is not repeated

Reply:
NVM overlooked this